pipeline/setup-runner.py

```diff
@@ -0,0 +1,330 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+#!/usr/bin/env python3
+"""Setup runner for Blender integration tests in CodeBuild."""
+import argparse
+import hashlib
+import os
+import platform
+import subprocess
+import sys
+import time
+from pathlib import Path
+
+BLENDER_VERSIONS = ["4.2.12", "4.5.4"]
+BLENDER_PYTHON_VERSIONS = {
+    "4.2.12": "3.11",
+    "4.5.4": "3.11",
+}
+USE_PUBLIC_URLS = False
+
+# SHA256 checksums from https://www.blender.org/download/
+BLENDER_CHECKSUMS = {
+    "4.2.12-linux-x64": "953717011e00a21bfd4ccf0e8af0d901b4c3ef09c48f14c16a18c146d858bcf7",
+    "4.5.4-linux-x64": "2e6ef8e99fc36327270429ddc8e7bad2859dd878a5a137d2e0bf0f02f6792505",
+    "4.2.12-windows-x64": "d7b77bf3a925722be87e5b5e429b584d7baa3bcc82579afa7952fc1f8c19d2e1",
+    "4.5.4-windows-x64": "0de55df1d99e4e7152605022cb648e795d5d49209c5c5c4889e1a19fb401a054",
+    "4.2.12-macos-arm64": "810bc64b89af7f9028b9d7544a34f32ad900ac6d913fd2f288895f10dc6c2527",
+    "4.5.4-macos-arm64": "7d6bd807563f0af65735cf9e21b788f6ac78bc5ceb87b96c424459785a13cd60",
+}
+
+
+def run(cmd, shell=False, check=True):
+    print(f"Running: {cmd if isinstance(cmd, str) else ' '.join(cmd)}")
+    result = subprocess.run(cmd, shell=shell)
+    if check and result.returncode != 0:
+        sys.exit(result.returncode)
+    return result
+
+
+def download_from_s3(s3_path, local_path):
+    bucket = os.environ.get("DEPENDENCY_BUCKET")
+    if not bucket:
+        print("DEPENDENCY_BUCKET not set")
+        return False
+    run(["aws", "s3", "cp", f"s3://{bucket}/{s3_path}", str(local_path), "--no-progress"])
+    return True
+
+
+def verify_checksum(file_path, expected_checksum):
+    """Verify SHA256 checksum of downloaded file."""
+    if not USE_PUBLIC_URLS:
+        return True
+
+    print(f"Verifying checksum for {file_path}...")
+    sha256 = hashlib.sha256()
+    with open(file_path, "rb") as f:
+        for chunk in iter(lambda: f.read(8192), b""):
+            sha256.update(chunk)
+
+    actual = sha256.hexdigest()
+    if actual != expected_checksum:
+        print("ERROR: Checksum mismatch!")
+        print(f"  Expected: {expected_checksum}")
+        print(f"  Actual:   {actual}")
+        sys.exit(1)
+
+    print("OK Checksum verified")
+    return True
+
+
+def setup_linux(python_version, install_x11=False):
+    pkg_mgr = (
+        "dnf"
+        if subprocess.run("command -v dnf", shell=True, capture_output=True).returncode == 0
+        else "yum"
+    )
+    run(f"{pkg_mgr} update -y", shell=True)
```

Comment:
As a best-practice, we should not be using `shell=True` and instead supply arguments as a list of `str`s. The reasons:

1. Avoids the need for an extra a shell process in between
2. Avoids needing to think about the proper use of escaping when arguments contains special shell characters (e.g. spaces, quotes, `*`)
3. It helps reduce the chance of a future command injection attack when we put a dynamic value into the command string

```suggestion
    run([pkg_mgr, "update", "-y"])
```

Same applies to the code below.

Reply:
Removed all cases of `shell=true`